plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -89,7 +89,42 @@ func LeastRequestedPriority(pod *api.Pod, podLister algorithm.PodLister, minionL
 
 	list := algorithm.HostPriorityList{}
 	for _, node := range nodes.Items {
-		list = append(list, calculateOccupancy(pod, node, podsToMachines[node.Name]))
+		list = append(list, calculateResourceOccupancy(pod, node, podsToMachines[node.Name]))
+	}
+	return list, nil
+}
+
+func min(l, r int64) (m int64) {
+	m = r
+	if l < r {
+		m = l
+	}
+	return m
+}
+
+// See comment for DumbSpreadingPriority()
+const dumbSpreadingDenominator int64 = 10
+
+// DumbSpreadingPriority is a priority function that favors nodes with fewer pods.
+// It works like LeastRequestedPeriority but instead of using 10 * percentage of machine free by resource,
+// it uses 10 * percentage of machine free by pod, with "percentage of machine free by pod" claculated as
+// (dumbSpreadingDenominator - number of pods already on the node + 1) / dumbSpreadingDenominator.
+// dumbSpreadingDenominator serves like the machine capacity in LeasRequestedPriority but is chosen
+// so that we equate one pod with a reasonable amount of resources when we combine all the scores together.
```

Comment:
It isn't clear why this is better than leastrequester with:
```go
if pod.resource == 0 {
  pod.resource = 10
}
```
One reason is that with the suggested pod.resource defaulting you can never have > 10 unspecified resource pods per node, while your algorithm will just stop contributing to node score in the >10 pods sitaution. 

However, I get the feeling we can end still end up with an imbalanced set of pods for exactly this reason. For example if I start off with an imbalanced distribution of cluster addons and I create a 100 pod rc in a limitless namespace, DumbSpreading will do the right thing till there are roughly 8-10 pods on each node, at which point whatever contribution it makes will not outweight the other 2.

This is why I initially considered dynamic weights, tell this algorithm to outweigh everything else in the no resource specified case and keep completely quiet otherwise. If I've misunderstood something and the situation described can never happen, please elaborate the comment.